document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoProjectionExpressionParser.java

```diff
@@ -10,14 +10,14 @@
 import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
 import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
 import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
-import org.hypertrace.core.documentstore.expression.type.SelectingExpression;
+import org.hypertrace.core.documentstore.expression.type.SelectTypeExpression;
 
 @Slf4j
-public final class MongoProjectionSelectingExpressionParser extends MongoSelectingExpressionParser {
+public final class MongoProjectionExpressionParser extends MongoSelectTypeExpressionParser {
```

Comment:
It feels like this class is parsing an expression called `ProjectionExpression`, however, there is no such expression class.
A few alternatives
* `MongoProjectingExpressionParser`
* `MongoProjectingSelectionParser`
* `MongoProjectingParser`
* `MongoProjectionParser`

Better to maintain the gerund (...ing) form, when the parser is doing something extra on top of the base parser.